lib/bundler/resolver.rb

```diff
@@ -324,8 +324,8 @@ def verify_gemfile_dependencies_are_found!(requirements)
             end
           else
             message = "Could not find gem '#{requirement}' "
-            if @index.source_types.include?(Bundler::Source::Rubygems)
-              message << "in any of the gem sources listed in your Gemfile."
+            if Bundler.definition.has_rubygems_remotes?
```

Comment:
OK. So, perhaps `Index#source_types` should be re-implemented?

I don't understand how to find the Gemfile's sources from `@index`.  I was thinking `@index.sources.each {|source| source.sources}` might be helpful, but after experimentation any source's source seems to always be empty.

Another simpler solution would be to remove this conditional completely so that the message here is always `in any of the gem sources listed in your Gemfile or installed on this machine.`.